File: Swift/excel-sheet-column-number.swift
class Solution {
    func titleToNumber(_ s: String) -> Int {
        var res = 0
        let valueA = "A".unicodeScalars.first!.value
        
        for scalar in s.unicodeScalars {
            res = res * 26 + Int(scalar.value - valueA) + 1
        }
        
        return res
    }
}

File: Python/excel-sheet-column-number.py
class Solution:
    def titleToNumber(self, s):
        """
        :type s: str
        :rtype: int
        """
        res = 0
        valueA = ord('A')

        for c in s:
            res = res * 26 + ord(c) - valueA + 1

        return res
